### ISSUE TITLE
amp responder changes int to string

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4064]
|<img alt="hagna's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| hagna reported|
|-|-|
|Trac ID|trac#4064|
|Type|defect|
|Created|2009-10-27 22:16:42Z|

Steps to duplicate

```
from twisted.trial.unittest import TestCase
from twisted.internet.defer import Deferred
from twisted.protocols import amp

class BugCMD(amp.Command):
    arguments = [('identifier', amp.String())]

    response = [('error', amp.Integer())]



class NetworkController(amp.AMP):
    def startBug2(self, identifier):
        return {'error':0}
    BugCMD.responder(startBug2)



class ControllerTests(TestCase):

    def setUp(self):
        self.controller = NetworkController()
        self.identifier = 123


    def test_BugCMD(self):

        responder = self.controller.lookupFunction(BugCMD.commandName)
        d = responder({
                'identifier': self.identifier,
                })

        def gotBugCMD(box):
            self.assertEqual(box, amp.AmpBox({'error':0}))
        d.addCallback(gotBugCMD)
        return d
```

<details><summary>Searchable metadata</summary>

```
trac-id__4064 4064
type__defect defect
reporter__hagna hagna
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__invalid invalid
component__core core
keywords__ 
time__1256681802000000 1256681802000000
changetime__1256682763000000 1256682763000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_1"></a>|@glyph set status to `closed`|
|-|-|

AmpBoxes are _always_ mappings of string to string, and the results of the return of lookupFunction will also be mapping only string to string.  The fact that this doesn't completely blow up and fail is an accident, which happens mostly because it would be too expensive to check the argument type of everything passed into an AmpBox.

In other words, you're passing bad arguments, but this is otherwise working as expected.  If you want to get an integer out, you're going to need to de-serialize your AMP results, using (in this case) `BugCMD.parseArguments(box, self.controller)`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
